docs/api/implementation-guide.md

```diff
@@ -0,0 +1,53 @@
+---
+
+# NuGet Server Implementation Guide
+
+Prior to 2019, NuGet package repositories could copy the package metadata from the `nupkg`'s [`nuspec` file](../reference/nuspec.md), and return the results unmodified in [search](./search-query-service-resource.md) and [package metadata](./registration-base-url-resource.md) requests.
+Since then, NuGet has evolved to provide developers with a richer experience, and this requires package repositories to do additional work in order to provide the additional value to developers.
+
+## Embedded files
+
+Package [icons](../reference/nuspec.md#icon), [license](../reference/nuspec.md#license), and [readme](../reference/nuspec.md#readme) files can be (and is recommended to be) embedded in the package.
+These files need a URL endpoint, either extracted and put on a static file, or a URL that dynamically extracts the files from the `.nupkg` on request, so that they can be viewed without downloading the entire `nupkg`.
+If your package repository provides package browsing and viewing package details, you can use the URLs to show customers the embedded content on your website.
+
+Finally, the [package metadata resource](./registration-base-url-resource.md) and [search resource](./search-query-service-resource.md) must contain the hosted URL in the `iconUrl`, `licenseUrl`, and/or `readmeUrl` properties of the JSON response.
+Packages (`.nupkg` files) are immutable and must not be modified.
+
+Note that the license could be an SPDX expression, or an embedded file (but not both).
+Packages that use a license expression, when represented in search and package metadata results, can have the `licenseUrl` set to the license expression, URL encoded, and appended to the end of https://licenses.nuget.org/.
+For example, https://licenses.nuget.org/Apache-2.0.
+
+## Owner field
+
+The [package manifest file (`.nuspec`)](../reference/nuspec.md) have two fields, `<authors>` and `<owners>`.
+Package authors who are packaging third-party content often put the third-party name in the `<authors>` field.
+The `<owners>` field is intended to denote who published the package on a repository, and therefore who should be contacted in case of packing issues or questions.
+
+The `<owners>` field has been deprecated from the `.nuspec` file, so if packages contain this metadata, it should be ignored.
+Do not return the value of the `.nuspec` file's `<owners>` field in the `owners` property in the search resource or package metadata resource JSON response.
+
+If your repository has per-package permissions, it is recommended to report the accounts that have permissions to publish new versions in the search and package metadata resources JSON responses.
+
+## Known vulnerability and deprecation data
+
+The [Package Metadata Resource](./registration-base-url-resource.md) can contain [deprecation](./registration-base-url-resource.md#package-deprecation) and [vulnerability](./registration-base-url-resource.md#vulnerabilities) information.
```

Comment:
@theot12 looking at the search resource docs, it doesn't contain deprecation or vulnerability info. Was this an oversight or is it intentional?

@donnie-msft @martinrrm I'm not sure what the status of PM UI no longer hitting registration to get deprecation and/or vulnerability info is, when the search endpoint returns the data. I'm not even sure if either of you are the right person to talk to about the client side of this feature.  Should we be documenting anything related to the search resource here, regarding deprecation and vulnerability information?

Reply:
From what I know, we are still using registration API to get vulnerabilities and deprecation information for all the versions. Search returns this information for the latest version only.

Reply:
We were holding off on documented depr+vuln in search until client actually does the implementation. We have flexibility to modify the protocol right now on NuGet.org since client doesn't use it (well, kind of... but yeah we have flexibility right now we would lose if we document the shape as is).